src/Lib/Canvas.vala

```diff
@@ -345,7 +348,14 @@ public class Akira.Lib.Canvas : Goo.Canvas {
             case EditMode.MODE_INSERT:
             case EditMode.MODE_SELECTION:
                 var selected_nob = nob_manager.selected_nob;
-                selected_bound_manager.transform_bound (event.x, event.y, selected_nob);
+                snap_manager.generate_snap_matches(selected_bound_manager.selected_items);
+                selected_bound_manager.transform_bound (event.x, event.y, selected_nob, snap_manager.snap_match_data);
+
+                if (snap_manager.snap_match_data.wants_snap()) {
+                    // if snapping occured, we need to update the grid before painting
+                    snap_manager.generate_snap_grid(selected_bound_manager.selected_items);
+                }
+                snap_manager.populate_decorators();
```

Comment:
Wouldn't be better to have all of this directly in the `SelectBoundManager` file?
Instead of getting the selected items from that file and then passing the `snap_match_data` to it, could we do all of this directly inside the `transform_bound()` method?
The `SelectBoundManager` file has the `Canvas` reference directly available on construct so you can tap the SnapManager easily.

Reply:
Yeah, this could be moved for sure. I wasn't sure what would be better.